src/react/components/pages/projectSettings/projectForm.json

```diff
@@ -3,7 +3,8 @@
     "properties": {
         "name": {
             "title": "${strings.common.displayName}",
-            "type": "string"
+            "type": "string",
+            "pattern": "^[^\\\\\\\\/:*?\\\\\\\"<>|]*$"
```

Comment:
Don't ask, don't tell... the trick was to escape this for both JSON and regex chars.  

Effectively won't allow a project name that contains: `<>"/|?*`